rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/RecoveryManager.java

```diff
@@ -82,6 +87,8 @@ public void persistParticipantInformation(final ParticipantInformation participa
             final Uid uid = new Uid(participantInformation.getId());
 
             recoveryStore.write_committed(uid, PARTICIPANT_INFORMATION_RECORD_TYPE, state);
+            // to identify the uid from the participant persisted into the object store in order to delete it later
+            persistedParticipants.put(participantInformation.getId(), uid);
         } catch (Exception e) {
             LOG.warn("Failure while persisting participant information.", e);
```

Comment:
@mmusgrov is it more or is the failure to write to disk of a participant just logged but ignored? Is that safe?

Reply:
Customers normally log at WARN level. Do you want it logged at ERROR instead or do you expect the code to recover from disk failures by retrying?

Reply:
I found I missed this comment. But the WARN level is considered because of the retrying happens and there is the hope the disk failure will be recovered.